src/shared/XLinkDispatcher.c

```diff
@@ -955,6 +958,29 @@ static int dispatcherClean(xLinkSchedulerState_t* curr)
     return 0;
 }
 
+static int dispatcherDeviceFdDown(xLinkSchedulerState_t* curr){
+
+    XLINK_RET_ERR_IF(pthread_mutex_lock(&reset_mutex), 1);
+    int ret = 0;
+
+    if (curr->dispatcherDeviceFdDown == 0) {
+
+        glControlFunc->closeDeviceFd(&curr->deviceHandle);
+        // Specify device FD was already closed
+        curr->dispatcherDeviceFdDown = 1;
+
+    } else {
+        ret = 1;
+    }
+
+    if(pthread_mutex_unlock(&reset_mutex) != 0) {
+        mvLog(MVLOG_ERROR, "Failed to unlock clean_mutex");
```

Comment:
Recommend changing error text to "reset_mutex"
Also, the `ret` value is `1` on ``unlock()` failure here. Yet in `dispatcherReset()` the same failure returns `0`. Is this inconsistent and/or an issue?

Reply:
> Also, the ret value is 1 on ``unlock()failure here. Yet indispatcherReset()` the same failure returns `0`. Is this inconsistent and/or an issue?

Not an issue - but added a `return 1;` as it'll do a WARN level log in that odd scenario (unlocking a mutex failing)